### PR TITLE
feat: client sdk - add support for DA publications

### PR DESCRIPTION
**Pull request by krzysu**

### CODE REVIEW

File: packages/client/src/publication/index.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 export type {
-  CreateCollectBroadcastItemResultFragment,
-  CreateCommentBroadcastItemResultFragment,
-  CreateMirrorBroadcastItemResultFragment,
-  CreatePostBroadcastItemResultFragment,
+  CreateCollectTypedDataFragment,
+  CreateCommentTypedDataFragment,
+  CreateMirrorTypedDataFragment,
+  CreatePostTypedDataFragment,
```
Comment (krzysu) [inline packages/client/src/publication/index.ts:5]: there is no need for keeping backward compatibility as these types are only introduced in #279

File: examples/node/scripts/createDAPostViaDispatcher.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    await lensClient.publication.createDataAvailabilityCommentViaDispatcher({
+      from: profileId,
+      contentURI,
+      commentOn: createPostResultValue.id,
+    });
+
+  const createCommentResultValue = createCommentResult.unwrap();
+
+  if (!isCreateDataAvailabilityPublicationResult(createCommentResultValue)) {
+    console.log(`Something went wrong`, createCommentResultValue);
```
Comment (bigint) [inline examples/node/scripts/createDAPostViaDispatcher.ts:60]: Do we need to consider this as template strings?

or we can make use of it

```suggestion
    console.log(`Something went wrong ${createCommentResultValue}`);
```

File: examples/node/scripts/createDAPostViaDispatcher.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    return;
+  }
+
+  console.log(`DA comment was created too: `, createCommentResultValue);
```
Comment (bigint) [inline examples/node/scripts/createDAPostViaDispatcher.ts:64]:
```suggestion
  console.log(`DA comment was created too: ${createCommentResultValue}`);
```

File: examples/node/scripts/createDAPostViaTypedData.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  const signedTypedData = await wallet._signTypedData(
+    createPostTypedDataValue.typedData.domain,
+    createPostTypedDataValue.typedData.types,
+    createPostTypedDataValue.typedData.value
+  );
+
+  console.log(`Broadcasting signed createDataAvailabilityPostTypedData...`);
+
+  const broadcastResult = await lensClient.transaction.broadcastDataAvailability({
+    id: createPostTypedDataValue.id,
```
Comment (bigint) [inline examples/node/scripts/createDAPostViaTypedData.ts:54]: I think we can destructure all variables 🙇🏼

Something like 

```
const { id, typedData } = createPostTypedDataResult.unwrap();
```

File: examples/node/scripts/createDAPostViaTypedData.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    signature: signedTypedData,
+  });
+
+  // broadcastResult is a Result object
+  const broadcastValue = broadcastResult.unwrap();
+
+  if (!isCreateDataAvailabilityPublicationResult(broadcastValue)) {
+    console.log(`Something went wrong`, broadcastValue);
```
Comment (bigint) [inline examples/node/scripts/createDAPostViaTypedData.ts:62]:
```suggestion
    console.log(`Something went wrong ${broadcastValue}`);
```

File: examples/node/scripts/createDAPostViaTypedData.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    return;
+  }
+
+  console.log(`DA post was created: `, broadcastValue);
```
Comment (bigint) [inline examples/node/scripts/createDAPostViaTypedData.ts:66]:
```suggestion
  console.log(`DA post was created: ${broadcastValue}`);
```

**Review by bigint — CHANGES_REQUESTED** [review]

File: examples/node/scripts/createDAPostViaDispatcher.ts (hunk 2 above)
Comment (krzysu) [reply examples/node/scripts/createDAPostViaDispatcher.ts:60]: no, it won't work, as I want to print out an object, not a string, so with your suggestion it ends up as `[object Object]`.  
Also please don't block a PR with "request changes" if you only want to suggest something insignificant in the scope of the PR.

compare
```
console.log(`Transaction status: `, wasIndexedFirstCheck.unwrap());
// vs.
console.log(`Transaction status: ${wasIndexedFirstCheck.unwrap()}`);
```

result
```
Transaction status:  {
  __typename: 'TransactionIndexedResult',
  indexed: false,
  txHash: '0xdfbf48b549813fc1c4654122132223d4ceccc9b1b6e71dbbec609560b4838fc0'
}
// vs.
Transaction status: [object Object]
```

File: examples/node/scripts/createDAPostViaTypedData.ts (hunk 4 above)
Comment (krzysu) [reply examples/node/scripts/createDAPostViaTypedData.ts:54]: this is an example script for integrators to understand how to use the SDK in different scenarios, I assumed the more meaningful names of consts/vars, the more self-explanatory the example is. I don't think restructuring makes any difference here, for sure it doesn't improve the clarity of the example. In the commented line of code, it is clear where the `id` is supposed to come from.

**Review by cesarenaldi — APPROVED** [review]
LGTM

**Pull request merged**